src/lualib/ArrayFlat.ts

```diff
@@ -1,7 +1,8 @@
 function __TS__ArrayFlat(this: void, array: any[], depth = 1): any[] {
     let result: any[] = [];
     for (const value of array) {
-        if (depth > 0 && type(value) === "table" && 1 in value) {
+        // A hack to concat only true "arrays" and empty tables.
```

Comment:
```suggestion
        // Workaround to determine if value is an array or not (fails in case of objects without keys)
        // See discussion in: https://github.com/TypeScriptToLua/TypeScriptToLua/pull/737
```
(Also change the other comment)